ui/ducks/ramps/ramps.ts

```diff
@@ -0,0 +1,58 @@
+import { createSelector } from 'reselect';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getCurrentChainId } from '../../selectors';
+import RampAPI from '../../helpers/ramps/rampApi/rampAPI';
+import { hexToDecimal } from '../../../shared/modules/conversion.utils';
+import { defaultBuyableChains } from './constants';
+import { AggregatorNetwork } from './types';
+
+export const fetchBuyableChains = createAsyncThunk(
+  'ramps/fetchBuyableChains',
+  async () => {
+    return await RampAPI.getNetworks();
+  },
+);
+
+const rampsSlice = createSlice({
+  name: 'ramps',
+  initialState: {
+    buyableChains: defaultBuyableChains,
+  },
+  reducers: {
+    setBuyableChains: (state, action) => {
+      state.buyableChains = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchBuyableChains.fulfilled, (state, action) => {
+        const networks = action.payload;
+        if (networks && networks.length > 0) {
+          state.buyableChains = action.payload;
```

Comment:
nit: in the assignment use the variable you are using in the condition
```suggestion
          state.buyableChains = networks;
```

Reply:
https://github.com/MetaMask/metamask-extension/commit/988e98509c129d77fb4127cbed2f94b974ee4718